girder/utility/filesystem_assetstore_adapter.py

```diff
@@ -270,3 +285,105 @@ def cancelUpload(self, upload):
         """
         if os.path.exists(upload['tempFile']):
             os.unlink(upload['tempFile'])
+
+    def importFile(self, item, path, user, name=None, mimeType=None, size=None):
+        """
+        Import a single file from the filesystem into the assetstore.
+
+        :param item: The parent item for the file.
+        :type item: dict
+        :param path: The path on the local filesystem.
+        :type path: str
+        :param user: The user to list as the creator of the file.
+        :type user: dict
+        :param name: Name for the file. Defaults to the basename of ``path``.
+        :type name: str
+        :param mimeType: MIME type of the file if known.
+        :type mimeType: str
+        :param size: Size of the file in bytes. Will stat the file if not
+            passed.
+        :type size: int
+        :returns: The file document that was created.
+        """
+        if size is None:
+            size = os.path.getsize(path)
+
+        name = name or os.path.basename(path)
+
+        file = self.model('file').createFile(
+            name=name, creator=user, item=item, reuseExisting=True,
+            assetstore=self.assetstore, mimeType=mimeType, size=size)
+        file['fullPath'] = os.path.abspath(os.path.expanduser(path))
+        file['imported'] = True
```

Comment:
I'm not sure if it makes sense to have such an endpoint as described in #961, as it breaks the transparency of the assetstore pattern. By design, the abstract assetstore adapter API doesn't provide any guarantees about hashing of the data, since that wasn't something we wanted to force on assetstore implementations. In fact we already have at least one adapter that does not even receive the bytes of the file, so computing a hash would be infeasible. If possible, I would prefer to keep content hashing as an internal implementation detail of our adapters.

Computing hashes of large files is also going to add a lot of time to a method like this, and I don't think it adds value -- in the case of imported files, we are not in charge of the file contents, so if they change on the underlying file, we don't care, we'll serve up whatever we're told, we just have to make sure the size is correct. For non-imported files, it shouldn't ever be the case that their content has changed unbeknownst to us.

It's my hope that whatever use case is meant to be fulfilled in #961 could also be handled using something other than hashes. If not, maybe we could create a custom plugin that will only work with certain assetstore types and exposes such an endpoint.